llvm/lib/Transforms/InstCombine/InstCombineAddSub.cpp

```diff
@@ -1775,6 +1775,15 @@ Instruction *InstCombinerImpl::visitAdd(BinaryOperator &I) {
     }
   }
 
+  // (add (add A, 1), (sext (icmp ne A, 0))) => call umax(A, 1)
+  if (match(LHS, m_OneUse(m_Add(m_Value(A), m_One()))) &&
```

Comment:
I think we should drop the one-use check on the add at least: https://llvm.godbolt.org/z/YGq6osY3z